bls-remote-signer/client/src/backend.rs

```diff
@@ -0,0 +1,91 @@
+use crate::api_error::ApiError;
+use crate::api_response::{PublicKeysApiResponse, SignatureApiResponse};
+use crate::rest_api::Context;
+use client_backend::BackendError;
+use hyper::Request;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_json::Value;
+use std::sync::Arc;
+
+lazy_static! {
+    static ref PUBLIC_KEY_FROM_PATH_REGEX: Regex = Regex::new(r"^/[^/]+/([^/]*)").unwrap();
+}
+
+/// HTTP handler to get the list of public keys in the backend.
+pub fn get_public_keys<T>(_: T, ctx: Arc<Context>) -> Result<PublicKeysApiResponse, ApiError> {
+    let public_keys = ctx
+        .backend
+        .clone()
```

Comment:
:oncoming_police_car: 

```
error[E0507]: cannot move out of an `Arc`
  --> bls-remote-signer/client/src/backend.rs:20:23
   |
20 |       let public_keys = ctx
   |  _______________________^
21 | |         .backend
   | |________________^ move occurs because value has type `backend::Backend<T>`, which does not implement the `Copy` trait
```

Reply:
Edit (5 hours later)

```diff --git a/bls-remote-signer/backend/src/lib.rs b/bls-remote-signer/backend/src/lib.rs
index 3ec25ca..c16872c 100644
--- a/bls-remote-signer/backend/src/lib.rs
+++ b/bls-remote-signer/backend/src/lib.rs
@@ -53,7 +53,7 @@ impl Backend<StorageRawDir> {
 
 impl<T: Storage> Backend<T> {
     /// Returns the available public keys in storage.
-    pub fn get_public_keys(self) -> Result<Vec<String>, BackendError> {
+    pub fn get_public_keys(&self) -> Result<Vec<String>, BackendError> {
         self.storage.get_public_keys()
     }
 
diff --git a/bls-remote-signer/client/src/backend.rs b/bls-remote-signer/client/src/backend.rs
index 87260cc..42a3c78 100644
--- a/bls-remote-signer/client/src/backend.rs
+++ b/bls-remote-signer/client/src/backend.rs
@@ -19,7 +19,6 @@ pub fn get_public_keys<T: Storage, U>(
 ) -> Result<PublicKeysApiResponse, ApiError> {
     let public_keys = ctx
         .backend
-        .clone()
         .get_public_keys()
         .map_err(|e| ApiError::ServerError(format!("{}", e)))?;
``` 

Should have `&self` the associated function...